### ISSUE TITLE
`command-${string}`-formatted `--success` condition fails with dashes

### ISSUE THREAD

**Issue by eventualbuddha** [open-cli-tools/concurrently#324]
This should use the exit code from the first command (`exit 1`), but doesn't because [the `successCondition` splits on `-`](https://github.com/open-cli-tools/concurrently/blob/706c4d6435cf4f1a3a92a533af0ced22c03e7865/src/completion-listener.ts#L54):
```sh
concurrently \
  --names 'this-one,that-one' \
  --success command-this-one \
  'exit 1' \
  'exit 0'
```

This version does work:
```sh
concurrently \
  --names 'this,that' \
  --success command-that \
  'exit 1' \
  'exit 0'
```

**Comment by gustavohenke** [comment]
Thanks for spotting this 🤦

**Comment by eventualbuddha** [comment]
Related, passing a command name that doesn't exist just silently ignores it. Maybe also true with indexes?

**Comment by paescuj** [comment]
> Related, passing a command name that doesn't exist just silently ignores it. Maybe also true with indexes?

Thanks for your input! We've discussed this in https://github.com/open-cli-tools/concurrently/pull/318#issuecomment-1125982525.

**Comment by gustavohenke** [comment]
Fix out in v7.2.1.
